src/test/java/pages/BasePage.java

```diff
@@ -7,27 +7,30 @@
 import org.openqa.selenium.support.ui.ExpectedConditions;
 import org.openqa.selenium.support.ui.WebDriverWait;
 import org.openqa.selenium.By;
+
 import java.time.Duration;
 
 public class BasePage {
     Actions actions;
     WebDriverWait wait;
     WebDriver driver;
```

Comment:
I recommend: make the variables protected, because protected access gives the subclass a chance to use the helper method or variable, while preventing a nonrelated class from trying to use it.

```
    protected Actions actions;
    protected WebDriverWait wait;
    protected WebDriver driver;
```